### PR TITLE
Convert Riot Pills to Slack mentions

### PR DESCRIPTION
**Pull request by Cadair**
This is a replacement for #53

### CODE REVIEW

File: lib/Main.js [hunk 1]
```diff
@@ -191,6 +191,10 @@ Main.prototype.getBotIntent = function() {
     return this._bridge.getIntent();
 };
 
+Main.prototype.getUsernamePrefix = function() {
```
Comment (Half-Shot) [inline lib/Main.js]: This should probably be "userid" prefix.

File: lib/substitutions.js [hunk 2]
```diff
@@ -134,6 +133,23 @@ var matrixToSlack = function(event, main) {
         if (event.content.msgtype == "m.emote") {
             string = "_" + string + "_";
         }
+
+        // replace riot "pill" behavior to "@" mention for slack users
+        var html_string = event.content.formatted_body;
```
Comment (Half-Shot) [inline lib/substitutions.js]: ``const`` please.

File: lib/substitutions.js [hunk 3]
```diff
@@ -134,6 +133,23 @@ var matrixToSlack = function(event, main) {
...
+        if (undefined != html_string) {
+            var regex = new RegExp('<a href="https://matrix.to/#/@' +
```
Comment (Half-Shot) [inline lib/substitutions.js]: ``const`` please.

File: lib/substitutions.js [hunk 4]
```diff
@@ -134,6 +133,23 @@ var matrixToSlack = function(event, main) {
...
+                                   main.getUsernamePrefix() +
+                                   '([^"]+)">([^<]+)</a>', "g");
+
+            var match = regex.exec(html_string);
```
Comment (Half-Shot) [inline lib/substitutions.js]: ``let`` please.

File: lib/substitutions.js [hunk 5]
```diff
@@ -134,6 +133,23 @@ var matrixToSlack = function(event, main) {
...
+            while (match != null) {
+                // Extract the slack ID from the matrix id
+                var userid = match[1].split("_")[1].split(":")[0];
```
Comment (Half-Shot) [inline lib/substitutions.js]: ``const`` please.

**Comment by Cadair** [comment]
What do you mean by config and schema update? This doesnt touch the config no?

**Comment by Cadair** [comment]
The username_prefix option is already in the config and the [schema](https://github.com/matrix-org/matrix-appservice-slack/blob/master/config/slack-config-schema.yaml#L38). I assume it dosent need changing?

**Comment by Half-Shot** [comment]
So it is, I thought you had added it. Looks good to me then :)

**Pull request merged**